ministryofjustice/laa-apply-for-criminal-legal-aid
# CRIMAP-691 Add employment status page

## Description of change
Adds employment status page behind feature flag
Add to income decision tree, but not in flow yet
Add route with alias, controller, form and errors
the non-js version of the checkbox exclusivity is not included in this PR
## Link to relevant ticket
https://dsdmoj.atlassian.net/browse/CRIMAP-691
## Notes for reviewer

## Screenshots of changes (if applicable)

### After changes:
<img width="717" alt="image" src="https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/assets/45827968/ee25a0fb-18a4-4fad-bdd3-83e2664e330b">
<img width="606" alt="image" src="https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/assets/45827968/677c8729-e185-4294-9891-e0f741e92f19">

## How to manually test the feature
run rails db:migrate
Start a new application
Complete client details
Go to /steps/income/what_is_clients_employment_status
Test validations by clicking save and continue without selecting a checkbox
Select any one or more checkboxes from the top section, and then by selecting My client is not working, the top checkboxes should clear (the non-js version is not included in this PR)
Check validation by selecting My client is not working but not selecting a radio and clicking save and continue
Select My client is not working, select No radio. Save and continue then click back into the application and come back to /steps/income/what_is_clients_employment_status - make sure text is autopopulated back into the text area
Select My client is not working, select Yes radio - should route to Did client lose job in custody page
